pkg/apis/walruscore/v1/resource_run.go

```diff
@@ -14,20 +15,67 @@ type ResourceRun struct {
 	meta.TypeMeta   `json:",inline"`
 	meta.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   ResourceDefinitionSpec   `json:"spec,omitempty"`
-	Status ResourceDefinitionStatus `json:"status,omitempty"`
+	Spec   ResourceRunSpec   `json:"spec,omitempty"`
+	Status ResourceRunStatus `json:"status,omitempty"`
 }
 
 var _ runtime.Object = (*ResourceRun)(nil)
 
 // ResourceRunSpec defines the desired state of ResourceRun.
 type ResourceRunSpec struct {
-	// TODO: your spec here
+	// Type is the type of the resource run.
+	//
+	// +k8s:validation:cel[0]:rule="oldSelf == self"
+	// +k8s:validation:cel[0]:message="immutable field"
+	Type ResourceRunType `json:"type"`
+
+	// DeployerType is the type of the deployer.
+	//
+	// +k8s:validation:cel[0]:rule="oldSelf == self"
+	// +k8s:validation:cel[0]:message="immutable field"
+	DeployerType string `json:"deployerType"`
```

Comment:
Whether the field should be placed in the status?

Reply:
replaced with templateFormat.